transcom/mymove
# B 18428 ssw form pdf work

## [B-18428](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A883374)

[INT PR](https://github.com/transcom/mymove/pull/12049)

## Summary

The Shipment Summary Worksheet is a file generated as part of the PPM Process. This task was to replace the existing png template images with a correctly filling PDF form template. This template can be viewed in Agility attachments for this story. This template was created in adobe, and then programmatically filled with a new service, SSWPPMGenerator. All new functions have unit tests written. A command and an API have both been updated to reflect the success of the generator, and will be elaborated on in testing instructions.

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

There are three tests that should be performed if possible, ranked from most to least important:

1. Access an SSW for an existing PPM Shipment. Requires a PPMShipment UUID.
The internal API Endpoint is: /ppm-shipments/{PPMShipmentID}/shipment_summary_worksheet?preparationDate=yyyy-mm-dd
So, an example would be:
http://milmovelocal:3000/internal/ppm-shipments/f093a13b-4ab8-4545-b24c-eb44bf52e605/shipment_summary_worksheet?preparationDate=2024-01-24
If you are logged in to MilMove as a User, and paste it into your URL with a correct PPMShipmentID from the database or your network tab under MTOShipments->ppmshipments->id, and a date, it should return the PDF to your browser.
This is also testable in SwaggerUI, but it will return PDF Code in the response viewer instead of to your browser, so please use the url bar.
Additionally, please be aware that many fields added in this version of the SSW are not yet available in the application. For a complete list of values and why they are unfilled, please review the Agility story.

2. Run the SSW Server test package, ensure all tests pass
These tests will test all new functions, as well as the form filling. It will fail if there is a corrupted or incorrectly configured PDF.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
